src/rust/lib/flexer/src/group/rule.rs

```diff
@@ -0,0 +1,32 @@
+use crate::automata::pattern::Pattern;
+
+
+
+/// A rule is a pair of regex pattern and callback.
+/// The intention is to run the callback after encountering given pattern.
```

Comment:
Add more info here - rules are matched top bottom, unlike in flex. This is important.

Reply:
Added to group.